metrics/providers/micrometer/src/main/java/io/helidon/metrics/providers/micrometer/MMeterRegistry.java

```diff
@@ -546,20 +625,34 @@ private <M extends Meter, HB extends MMeter.Builder<?, M, HB, HM>, HM extends MM
 
         io.helidon.metrics.api.Meter.Id id = mBuilder.id();
 
-        MMeter<?> foundMeter = metersById.get(id);
-        if (foundMeter != null) {
-            if (!mBuilder.meterType().isInstance(foundMeter)) {
-                throw new IllegalArgumentException("Attempt to get or create a meter of type "
-                                                           + mBuilder.meterType().getName()
-                                                           + " when an existing meter " + id
-                                                           + " has an incompatible type " + foundMeter.getClass()
-                        .getName());
+        lock.readLock().lock();
+
+        try {
+            MMeter<?> foundMeter = meterIfRegistered(mBuilder, id);
+            if (foundMeter != null) {
+                return (HM) foundMeter;
             }
-            return (HM) foundMeter;
+        } finally {
+            lock.readLock().unlock();
         }
 
-        lock.lock();
+        /*
+         "Promote" our lock from read to write.
```

Comment:
…although you don't have any lock at this point?

Reply:
Well, not at that moment--hence the quotes around "promote" because in fact just a few lines above we _had_ acquired the read lock and the effect we want is to promote the lock from read to write as the comment a few lines below explains. 

I've rewritten the comments to remove any possible confusion.

Reply:
LGTM. Acquiring WL and then checking condition again to account for the interim no lock was held.